libkbfs/tlf_journal.go

```diff
@@ -814,6 +817,47 @@ func (j *tlfJournal) convertMDsToBranchIfOverThreshold(ctx context.Context) (
 	return true, nil
 }
 
+func (j *tlfJournal) doOnMDFlush(ctx context.Context,
+	rmds *RootMetadataSigned) error {
+	if j.onMDFlush != nil {
+		j.onMDFlush.onMDFlush(rmds.MD.TlfID(), rmds.MD.BID(),
+			rmds.MD.RevisionNumber())
+	}
+
+	// Remove saved blocks in chunks to avoid starving foreground file
```

Comment:
Part of the problem might be the fact that we append to both journals when we have blocks saved. That in turn is necessary because we turn off GC for all blocks when we have blocks saved, when we should really be doing so only for the saved blocks. Maybe have a TODO/issue for that?